src/Drivers/Gd/Modifiers/TrimModifier.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Intervention\Image\Drivers\Gd\Modifiers;
+
+use Intervention\Image\Drivers\DriverSpecialized;
+use Intervention\Image\Exceptions\NotSupportedException;
+use Intervention\Image\Interfaces\ImageInterface;
+use Intervention\Image\Interfaces\ModifierInterface;
+
+/**
+ * @property int $tolerance
+ */
+class TrimModifier extends DriverSpecialized implements ModifierInterface
+{
+    public function apply(ImageInterface $image): ImageInterface
+    {
+        if ($image->isAnimated()) {
+            throw new NotSupportedException('Animated Images do not support trim operation');
+        }
+
+        $gd      = $image->core()->native();
+        $width   = imagesx($image->core()->native());
+        $height  = imagesy($image->core()->native());
+        $red     = 0;
+        $green   = 0;
+        $blue    = 0;
+        $corners = [
+            [0,0],
+            [$width - 1,0],
+            [0,$height - 1],
+            [$width - 1,$height - 1],
+        ];
+
+        foreach ($corners as $corner) {
+            $thisColor = imagecolorat($gd, $corner[0], $corner[1]);
+
+            $rgb    = imagecolorsforindex($gd, $thisColor);
+            $red   += round(round(($rgb['red'] / 0x33)) * 0x33);
+            $green += round(round(($rgb['green'] / 0x33)) * 0x33);
+            $blue  += round(round(($rgb['blue'] / 0x33)) * 0x33);
+        }
+
+        $red   /= 4;
+        $green /= 4;
+        $blue  /= 4;
+
+        $trimColor = imagecolorallocate($gd, (int)$red, (int)$green, (int)$blue);
+        $trimmed   = imagecropauto($gd, IMG_CROP_THRESHOLD, $this->tolerance, $trimColor);
```

Comment:
Basically it works. However, above certain tolerance values the function returns `false`. I think this should be handled.  For example, the test image "trim.png" shows an error from tolerance `67`.

I think it should be possible, as with Imagick, to enter values from 0 to 100 without errors occurring.

If a very high tolerance value is entered in Imagick, the result is a 1 x 1 pixel image.

Reply:
Do you want to limit the tolerance to 50?

Reply:
> Do you want to limit the tolerance to 50?

No, I would like to be able to set the parameter value between 0 and 100 and get the same result in GD and Imagick. Of course this is not 100% possible, but we should get as close as possible.

Reply:
> Meanwhile - do we really need to place limitations, and safeguards? I'm more inclined to set no limits for tolerance, and add a note in userdocs. How people choose to use the lib - is up to them?

Yes. I agree.

Reply:
Do you want to merge as is, I can fill in some more extreme use cases in the following days? :)

Reply:
Ok, let me know what do you want me to do :)

Reply:
> Ok, let me know what do you want me to do :)

Intervention Image is a standardized API for GD and Imagick. The main goal is to make the results of the API independent of the driver used and thus ensure interchangeability.

Of course, this goal is not always 100% achievable, and especially in this case it will not work, because both drivers work completely different. However, I have always done my best to get as close to 100% as possible. I expect the same for merge requests.

We have a new method that takes a parameter. The parameter can be passed values from 0 to 100. 

- The result of each of these values should be an instance of ImageInterface.
- The modification of the image should be as similar as possible with both drivers, same images and parameters.

I think the output of `trim()` with Imagick can be used as a model.

If you have any questions, feel free to contact me.